### PR TITLE
ASoC: SOF: ipc4-topology: set dmic dai index from copier

### PR DESCRIPTION
**Pull request by juimonen**
Dmic dai index was set incorrectly to bits 5-7, when it is actually using just all lowest 8 bits. As the dmic index can only be 0 or 1, just OR copier dai index directly to node_id and remove the macro for fiddling the dmic index.

Signed-off-by: Jaska Uimonen <jaska.uimonen@linux.intel.com>

### CODE REVIEW

**Comment by juimonen** [comment]
needs to be synced with https://github.com/thesofproject/sof/pull/6877. for windows compatibility

**Comment by juimonen** [comment]
the reason for this: https://github.com/thesofproject/sof/pull/6872

**Review by plbossart — DISMISSED** [review]

**Review by ranj063 — DISMISSED** [review]

**Comment by ranj063** [comment]
SOFCI TEST

**Review by singalsu — DISMISSED** [review]

**Comment by juimonen** [comment]
@libinyang can you test this and the related fw revert with your wov pipeline? you are probably currently the only one with that setup and this will affect you, thanks!

File: sound/soc/sof/ipc4-topology.c [hunk 1]
```diff
@@ -563,8 +563,7 @@ static int sof_ipc4_widget_setup_comp_dai(struct snd_sof_widget *swidget)
 		break;
 	case SOF_DAI_INTEL_DMIC:
 		/* set DMIC DAI index as the node_id */
-		ipc4_copier->data.gtw_cfg.node_id |=
-			SOF_IPC4_NODE_INDEX_INTEL_DMIC(ipc4_copier->dai_index);
+		ipc4_copier->data.gtw_cfg.node_id |= ipc4_copier->dai_index;
```
Comment (ujfalusi) [inline sound/soc/sof/ipc4-topology.c]: Do you have a fixes tag with SHA?

Comment (juimonen) [reply sound/soc/sof/ipc4-topology.c]: ok let me add

Comment (juimonen) [reply sound/soc/sof/ipc4-topology.c]: @ujfalusi added fixes tag

**Review by abonislawski — DISMISSED** [review]
This is good but if you prefer you can stay with SOF_IPC4_NODE_INDEX_INTEL_DMIC, just bit calculations must be correct.

```
typedef union _DmicDmaNodeIdVindex
{
    //! Raw 8-bit value of virtual index
    uint8_t b;
    //! Bit fields
    struct /*Bits*/
    {
        //! Index of the input queue within the DMIC instance.
        uint8_t dmic_interface_queue_id : 3;
        //! Index of the time slot group within the queue.
        uint8_t time_slot_group_index   : 2;
        //! Index of DMIC instance.
        uint8_t instance                : 3;
    } f; //!< Bits
} DmicDmaNodeIdVindex;
```

we need to set **dmic_interface_queue_id** instead of **instance** in current code

_time_slot_group_index_ and _instance_ are always 0

**Comment by juimonen** [comment]
updated v2. Perhaps we do it as @abonislawski suggested. Never know if we need to use the other bits in the future, and the change is even simpler/smaller.

**Review by abonislawski — APPROVED** [review]

**Review by ujfalusi — APPROVED** [review]

**Review by plbossart — APPROVED** [review]

**Review by ranj063 — APPROVED** [review]

**Pull request merged**